### PR TITLE
[ISSUE #1015] HttpUrlConnection based proxy configuration

### PR DESCRIPTION
**Pull request by LuanLouis**
## What is the purpose of the change

HttpClient supports proxy configuration

## Brief changelog

Add proxy related configuration:
```java
public static final Boolean CON_PROXY_ENABLED =  Boolean
        .getBoolean("com.alibaba.nacos.client.naming.proxy.enable");
    public static final String CON_PROXY_HOST = System.
        getProperty("com.alibaba.nacos.client.naming.proxy.host", "127.0.0.1");
    public static final Integer CON_PROXY_PORT = Integer.
        getInteger("com.alibaba.nacos.client.naming.proxy.port", 8001);
```
if proxy is enabled,`HttpClient` proxy is configured like this:
```java
public static HttpResult request(String url, List<String> headers, Map<String, String> paramValues, int connectTimeout, int readTimeout, String encoding, String method) {
        HttpURLConnection conn = null;
        try {
            String encodedContent = encodingParams(paramValues, encoding);
            url += (null == encodedContent) ? "" : ("?" + encodedContent);
            //check if proxy is enabled
            if(CON_PROXY_ENABLED){
                InetSocketAddress proxyAddress = new InetSocketAddress(CON_PROXY_HOST,CON_PROXY_PORT);
                Proxy proxy = new Proxy(Proxy.Type.HTTP, proxyAddress);
                conn = (HttpURLConnection) new URL(url).openConnection(proxy);
            }else{
                conn = (HttpURLConnection) new URL(url).openConnection();
            }
            conn.setConnectTimeout(connectTimeout);
            conn.setReadTimeout(readTimeout);
            conn.setRequestMethod(method);

            conn.addRequestProperty("Client-Version", UtilsAndCommons.SERVER_VERSION);
            conn.addRequestProperty("User-Agent", UtilsAndCommons.SERVER_VERSION);
            setHeaders(conn, headers, encoding);
            conn.connect();

            return getResult(conn);
        } catch (Exception e) {
            Loggers.SRV_LOG.warn("Exception while request: {}, caused: {}", url, e);
            return new HttpResult(500, e.toString(), Collections.<String, String>emptyMap());
        } finally {
            if (conn != null) {
                conn.disconnect();
            }
        }
    }
```

### CODE REVIEW

**Comment by LuanLouis** [comment]
#1015 HttpClient doesn't support proxy configuration

File: client/src/main/java/com/alibaba/nacos/client/naming/net/HttpClient.java [hunk 1]
```diff
@@ -71,7 +74,13 @@ public static HttpResult request(String url, List<String> headers, Map<String, S
             String encodedContent = encodingParams(paramValues, encoding);
             url += (StringUtils.isEmpty(encodedContent)) ? "" : ("?" + encodedContent);
 
-            conn = (HttpURLConnection) new URL(url).openConnection();
+            if(CON_PROXY_ENABLED){
```
Comment (nkorange) [inline client/src/main/java/com/alibaba/nacos/client/naming/net/HttpClient.java]: I think the code is not formatted.

File: naming/src/main/java/com/alibaba/nacos/naming/misc/HttpClient.java [hunk 2]
```diff
@@ -103,7 +102,13 @@ public static HttpResult request(String url, List<String> headers, Map<String, S
             String encodedContent = encodingParams(paramValues, encoding);
             url += (null == encodedContent) ? "" : ("?" + encodedContent);
 
-            conn = (HttpURLConnection) new URL(url).openConnection();
+            if(CON_PROXY_ENABLED){
```
Comment (nkorange) [inline naming/src/main/java/com/alibaba/nacos/naming/misc/HttpClient.java]: This HttpClient is just for naming module, can you add the same proxy setting for config module?

**Comment by nkorange** [comment]
Close this PR due to no feedback from author.

**Pull request closed without merging**